src/main/java/com/rapid7/logback/LogentriesAppender.java

```diff
@@ -171,18 +176,22 @@ public void setLogID(String logID) {
     }
 
     /**
-     * Sets the suffixPattern to be the pattern field in the .xml configuration file
+     * Sets the encoder for this appender
      *
-     * @param encoder Logback Pattern Layout Encoder
+     * @param encoder Logback Encoder
      */
-    public void setEncoder(PatternLayoutEncoder encoder) {
-        this.suffixPattern = encoder.getPattern();
+    public void setEncoder(Encoder<ILoggingEvent> encoder) {
+        this.encoder = encoder;
     }
 
     @Override
     public void start() {
-        if (layout == null) {
-            layout = buildLayout();
+        if (encoder == null) {
+            layout = layout == null ? buildLayout() : layout;
+            LayoutWrappingEncoder lwe = new LayoutWrappingEncoder<ILoggingEvent>();
```

Comment:
Probably should be in this form?

```suggestion
            LayoutWrappingEncoder<ILoggingEvent> lwe = new LayoutWrappingEncoder<>();
```